cobc/codegen.c

```diff
@@ -2575,40 +2650,118 @@ output_collating_tables (void)
+	if (gen_colseq_native_ebcdic) {
+		output_storage ("\n/* Native to EBCDIC table for COLLATING SEQUENCE */\n");
+		output_colseq_table_empty ("cob_colseq_native_ebcdic");
+		output_storage ("\n");
+	}
+	if (gen_colseq_native_ascii) {
+		output_storage ("\n/* Native to ASCII table for COLLATING SEQUENCE */\n");
+		output_colseq_table_empty ("cob_colseq_native_ascii");
+		output_storage ("\n");
+	}
+
+	/* Charset-aware conversion tables for CONVERT */
+	if (gen_convert_ebcdic_ascii) {
+		output_storage ("\n/* EBCDIC to ASCII table for CONVERT */\n");
+		output_colseq_table_empty ("cob_convert_ebcdic_ascii");
+		output_colseq_table_field("f_convert_ebcdic_ascii", "cob_convert_ebcdic_ascii");
+		output_storage ("\n");
+	}
+	if (gen_convert_ascii_ebcdic) {
+		output_storage ("\n/* ASCII to EBCDIC table for CONVERT */\n");
+		output_colseq_table_empty ("cob_convert_ascii_ebcdic");
+		output_colseq_table_field("f_convert_ascii_ebcdic", "cob_convert_ascii_ebcdic");
+		output_storage ("\n");
+	}
+
+	/* Charset-aware conversion tables for CODE-SET */
+	if (gen_codeset_ascii_native) {
+		output_storage ("\n/* ASCII to Native table for CODE-SET */\n");
+		output_colseq_table_empty ("cob_codeset_ascii_native");
+		output_storage ("\n");
+	}
+	if (gen_codeset_native_ascii) {
+		output_storage ("\n/* Native to ASCII table for CODE-SET */\n");
+		output_colseq_table_empty ("cob_codeset_native_ascii");
+		output_storage ("\n");
+	}
+	if (gen_codeset_ebcdic_native) {
+		output_storage ("\n/* EBCDIC to Native table for CODE-SET */\n");
+		output_colseq_table_empty ("cob_codeset_ebcdic_native");
+		output_storage ("\n");
+	}
+	if (gen_codeset_native_ebcdic) {
+		output_storage ("\n/* Native to EBCDIC table for CODE-SET */\n");
+		output_colseq_table_empty ("cob_codeset_native_ebcdic");
+		output_storage ("\n");
+	}
 }
 
 static void
 output_init_collating_tables (void)
 {
-	if ((gen_ascii_ebcdic > 0) || (gen_ebcdic_ascii > 0)) {
+	/* Generic conversion tables */
+	if (gen_ascii_ebcdic > 0 || gen_ebcdic_ascii > 0) {
 		output_line ("cob_get_collation_by_name(\"%s\", %s, %s);",
-				cob_get_collation_name(cb_ebcdic_table),
-				(gen_ebcdic_ascii > 0) ? "&cob_ebcdic_ascii" : "NULL",
-				(gen_ascii_ebcdic > 0) ? "&cob_ascii_ebcdic" : "NULL");
-		if (gen_ascii_ebcdic > 1) {
-			output_line("f_ascii_ebcdic.data = (cob_u8_ptr)cob_ascii_ebcdic;");
-		}
+			     cob_get_collation_name(cb_ebcdic_table),
+			     gen_ebcdic_ascii > 0 ? "&cob_ebcdic_ascii" : "NULL",
+			     gen_ascii_ebcdic > 0 ? "&cob_ascii_ebcdic" : "NULL");
 		if (gen_ebcdic_ascii > 1) {
 			output_line("f_ebcdic_ascii.data = (cob_u8_ptr)cob_ebcdic_ascii;");
 		}
+		if (gen_ascii_ebcdic > 1) {
+			output_line("f_ascii_ebcdic.data = (cob_u8_ptr)cob_ascii_ebcdic;");
+		}
+	}
+
+	/* Charset-aware conversion tables for COLLATING SEQUENCE */
+	if (gen_colseq_native_ebcdic > 0 || gen_colseq_native_ascii > 0) {
+		output_line ("cob_build_collation(\"%s\", \"%s\", \"%s\", %s, %s);",
```

Comment:
The return code of this function is ignored - should this be done? Similar below.